### PR TITLE
(dev/core#2994) IFrame Connector - Add extension to allow remote iframes (*incubation*)

### PR DESCRIPTION
**Pull request by totten**
Overview
----------------------------------------

The humble `<iframe>` has been around since HTML 4, and it provides a widely-known mechanism to embed content from other web-sites. However, modern `<iframe>`s are more complicated than their ancestors -- they require additional adaptations to operate with current browsers. And (at the CMS layer) `<iframe>` support is often disabled by default.

This PR provides opt-in support for `<iframe>`s for CiviCRM. By enabling the extension ("IFrame Connector"), one can allow external web-sites to embed CiviCRM pages.

* This is a key building-block for oEmbed support (https://lab.civicrm.org/dev/core/-/issues/2994). oEmbed providers often use IFRAMEs.
* The extension should be regarded as *incubating*. The PR has the main concepts. Both @seamuslee001 and I have been able to use it. It has connectors for `UF=Drupal` and `UF=Drupal8`. But we need to develop similar connectors for other UFs, and we will probably revise the options further in the coming months. Therefore, it is hidden (`<tag>mgmt:hidden</tag>`) from the main list of extensions.
* This is a re-roll of #29075. It's rebased. Debug comments have been removed. The README and `info.xml` have been cleaned up more. The extension has been renamed(`oembed`=>`iframe`) from earlier drafts.

Before
----------------------------------------

* In ancient times, you could kind-of put CiviCRM content in an IFRAME.
* In modern times, it's not very reliable. There are traps.

After
----------------------------------------

* You can enable the "IFrame Connector" extension. And then certain pages (e.g. public-facing pages) can be embedded.

Quick Start
----------------------------------------

* Enable the extension via CLI/API:

    ```bash
    cv en iframe
    ````

* The "System Status" will prompt you to deploy the connector script:

    Screenshot 2024-02-23 at 8 52 50 PM

* In "Administer > System Settings > IFrame Connector", you can configure how embedded pages will work:

    Screenshot 2024-02-23 at 8 44 42 PM

* Pick a public CiviCRM page (eg `civicrm/contribute/transact?reset=1&id=1`).
* On an external website, you can make an HTML page which embeds the CiviCRM page:
    ```html
    <IFRAME SRC="http://example.org/iframe.php/civicrm/contribute/transact?reset=1&id=1"/>
    ```

Comments
----------------------------------------

The included `ext/iframe/README.md` includes more, discussion, and list of TODOs.

Why rename the extension? In retrospect, the current code is actually about IFRAMEs. IFRAMEs can be used with or without oEmbed. The follow-up will provide an `oembed` extension which is actually about [oEmbed](https://oembed.com/) (i.e. the protocol for translating copy-pasted URLs into IFRAME snippets). And `oembed` will depend on `iframe`.

### CODE REVIEW

**Comment by totten** [comment]
There's currently one test failure (`api\v4\Action\ContactGetTest.testAge`). I don't think this is related. At least, it's hard to see a connection, and the same error appears on a different/concurrent PR ([test-run 9400/PR 29496/test-3](https://test.civicrm.org/job/CiviCRM-Core-Matrix-PR/9400/BKPROF=dfl,SUITES=phpunit-api4,label=bknix-tmp/) vs [test-run 9401/#29504/test-4](https://test.civicrm.org/job/CiviCRM-Core-Matrix-PR/9401/BKPROF=dfl,SUITES=phpunit-api4,label=bknix-tmp/)).

**Comment by seamuslee001** [comment]
Yes I can confirm I have been able to generate a suitable iframe where d10 is the provider and a WP is consumer so far

File: .gitignore [hunk 1]
```diff
@@ -22,6 +22,7 @@
 !/ext/financialacls
 !/ext/contributioncancelactions
 !/ext/recaptcha
+!/ext/oembed
```
Comment (seamuslee001) [inline .gitignore:25]: @totten it seems you have re-named the ext to be iframe but this is still set to be oembed?

Comment (totten) [reply .gitignore]: Thanks @seamuslee001. Fixed.

**Comment by colemanw** [comment]
Reviewed the code, tried out the functionality. Good to merge.

**Comment by totten** [comment]
Failures are a Leap Day Present.

**Comment by demeritcowboy** [comment]
Every page in drupal 8 is now giving php warnings - followup PR: https://github.com/civicrm/civicrm-core/pull/29560

**Pull request merged**